src/test/java/edu/ucsb/cs156/example/controllers/RecommendationControllerTests.java

```diff
@@ -0,0 +1,377 @@
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                delete("/api/Recommendation?id=15")
+                                                .with(csrf()))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+                verify(recommendationRepository, times(1)).findById(15L);
+                verify(recommendationRepository, times(1)).delete(any());
+
+                Map<String, Object> json = responseToJson(response);
+                assertEquals("Recommendation with id 15 deleted", json.get("message"));
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_tries_to_delete_non_existant_recommendation_and_gets_right_error_message()
+                        throws Exception {
+                // arrange
+
+                when(recommendationRepository.findById(eq(15L))).thenReturn(Optional.empty());
+
+                // act
+                MvcResult response = mockMvc.perform(
+                                delete("/api/Recommendation?id=15")
+                                                .with(csrf()))
+                                .andExpect(status().isNotFound()).andReturn();
+
+                // assert
+                verify(recommendationRepository, times(1)).findById(15L);
+                Map<String, Object> json = responseToJson(response);
+                assertEquals("Recommendation with id 15 not found", json.get("message"));
+        }
+
+        @WithMockUser(roles = { "ADMIN", "USER" })
+        @Test
+        public void admin_can_edit_an_existing_recommendation() throws Exception {
+                // arrange
+
+                LocalDateTime dR1 = LocalDateTime.parse("2022-01-03T00:00:00");
+                LocalDateTime dN1 = LocalDateTime.parse("2022-02-03T00:00:00");
+
+                Recommendation recommendationOrig = Recommendation.builder()
+                                .requesterEmail("jgaucho@ucsb.edu")
+                                .professorEmail("teacher@ucsb.edu")
+                                .explanation("bad lectures")
+                                .dateRequested(dR1)
+                                .dateNeeded(dN1)
+                                .done(false)
+                                .build();
+                
+                LocalDateTime dR2 = LocalDateTime.parse("2022-01-04T00:00:00");
+                LocalDateTime dN2 = LocalDateTime.parse("2022-02-05T00:00:00");
+
+                Recommendation recommendationEdited = Recommendation.builder()
+                                .requesterEmail("johnjoe@ucsb.edu")
+                                .professorEmail("prof@ucsb.edu")
+                                .explanation("more homework")
+                                .dateRequested(dR2)
+                                .dateNeeded(dN2)
+                                .done(false)
```

Comment:
Also, make sure that the `done` parameter is different between `recommendationOrig` and `recommendationEdited` so that it covers the mutation coverage.

Reply:
fixed in most recent commit